### ISSUE TITLE
GraphQlClient Feedback

### ISSUE THREAD

**Issue by WhiteCat22** [OpenLiberty/guide-graphql-client#18]
Instead of "resolvers" I would refer your interface methods as "queries" and "mutations".

Best practices would be to explicitly label all of your queries with `@Query` and all of your mutations with `@Mutation`. You should also note that the name of the query or mutation is the method name but can be overridden with the `@Query` or `@Mutation` annotations.

Same with the `@Name` annotation, graphql inputs will be the variable name unless overridden by `@Name`.

Example:
```
@GraphQLClientApi
public interface GraphQlClient {

    // the method name is "system" but the query will be named "SystemInfo"
    @Query("SystemInfo")
    SystemInfo system(@Name("hostname") String host);  // same thing here, variable is "host" but the query variable will be "hostname"

}
```

I suggest using different method names and annotation names to demonstrate how the name overriding works.

**Comment by gkwan-ibm** [comment]
hi @WhiteCat22 

It does not work if using `@Query("SystemInfo")` but it works if we using ` @Query("system")` that is same as the method and that matches to the [graphql](https://github.com/OpenLiberty/draft-guide-graphql-client/blob/draft/finish/graphql/src/main/java/io/openliberty/guides/graphql/GraphQLService.java) service

**Comment by gkwan-ibm** [comment]
Do you think following is better?
```
    @Query("system")
    SystemInfo getSystemInfo(@Name("hostname") String host);

    @Query("systemLoad")
    SystemLoad[] getSystemLoad(@Name("hostnames") String[] hostnames);
```

**Comment by WhiteCat22** [comment]
Ah, I was just providing an example of how you can name the query something other than the method name.

I've not used the SmallRye Graphql client much so you might have to call the actual method name with the client, wheres the actual graphql request that goes over the wire will use the `@Query` name.

If you got to `http://<HOSTNAME>:<PORT>/graphql/schema.graphql` you can use the built in GraphiQL interface to send GraphQL requests to the application and you will see that it uses the name defined by the `@Query` annotation.

Is the purpose of this guide to suppliment https://openliberty.io/guides/microprofile-graphql.html#running-graphql-queries and show how to use the Smallrye GraphQL Client with that application?

**Comment by gkwan-ibm** [comment]
- >I've not used the SmallRye Graphql client much so you might have to call the actual method name with the client, wheres the actual graphql request that goes over the wire will use the @Query name.

  - Is there other better graphql client that we should use? Or SmallRye Graphql client is fine to use?

` `

- >Is the purpose of this guide to suppliment https://openliberty.io/guides/microprofile-graphql.html#running-graphql-queries and show how to use the Smallrye GraphQL Client with that application?
  - yes

**Comment by shin19991207** [comment]
Addressed in #19

**Comment by WhiteCat22** [comment]
Looks Great!
